tests/test_models.py

```diff
@@ -25,10 +10,116 @@ def openai_mock_client():
 ]
 
 
-def test_openai_model(openai_mock_client):
-    for model_name in list(OpenAIModel.MODELS) + list(OpenAIModel.SHORTCUTS):
-        TEST_MODEL_ARGUMENTS = ModelArguments(model_name)
-        with patch("sweagent.agent.models.config.Config"), patch("sweagent.agent.models.OpenAI"):
-                model = OpenAIModel(TEST_MODEL_ARGUMENTS, [])
-        model.client = openai_mock_client
+@pytest.fixture
+def mock_openai_response():
+    choice = Mock(message=Mock(content="test"))
+    usage = Mock(prompt_tokens=10, completion_tokens=10)
+    return Mock(choices=[choice], usage=usage)
+
+
+@pytest.fixture
+def mock_anthropic2_response():
+    return Mock(
+        id="test",
+        completion="Hello. I'm fine.",
+        model="claude-haiku",
+        stop_reason=None,
+        type="completion",
+        usage=Mock(input_tokens=10, output_tokens=10)
+    )
+
+
+@pytest.fixture
+def mock_anthropic3_response():
+    return Mock(
+        id="test",
+        content=[Mock(type="text", text="Hello, I'm fine")],
+        model="claude-haiku",
+        role="assistant",
+        stop_reason=None,
+        stop_sequence=None,
+        type="message",
+        usage=Mock(input_tokens=10, output_tokens=10)
+    )
+
+
+@pytest.fixture
+def mock_ollama_response():
+    return {
+        "prompt_eval_count": 10,
+        "eval_count": 10,
+        "message": {
+            "content": "Hello. I'm fine."
+        }
+    }
+
+
+def split_claude_model_by_version():
```

Comment:
For Claude-2, we were sending requests with `completions.create()`, while for Claude-3, we were sending requests with `messages.create()`, so we needed to separate the test cases and divide the models.